### PR TITLE
streamingccl: support sslinline for replication URLs

### PR DESCRIPTION
**Pull request by stevendanna**
The lib/pq library supports the non-standard sslinline feature. We lost this support when moving to pgx. This adds basic sslinline support to the partition stream client.

This expects relevant certificate data is URL encoded.

Epic: CRDB-18749

Release note: None

### CODE REVIEW

File: pkg/ccl/streamingccl/streamclient/partitioned_stream_client.go [hunk 1]
```diff
@@ -358,3 +377,50 @@ func (p *partitionedStreamSubscription) Events() <-chan streamingccl.Event {
...
+	v := ret.Query()
+	v.Del("sslcert")
+	v.Del("sslkey")
+	v.Del("sslrootcert")
+	v.Del("sslinline")
+	ret.RawQuery = v.Encode()
+	return &ret
+}
+
+func AddTLSCertsToConfigFromInlineURL(remote *url.URL, tlsConfig *tls.Config) error {
```
Comment (dt) [inline pkg/ccl/streamingccl/streamclient/partitioned_stream_client.go:392]: I'm guessing there's some expected future use of these outside the package requiring them to be exported?

File: pkg/ccl/streamingccl/streamclient/partitioned_stream_client.go [hunk 2]
```diff
@@ -358,3 +377,50 @@ func (p *partitionedStreamSubscription) Events() <-chan streamingccl.Event {
 func (p *partitionedStreamSubscription) Err() error {
 	return p.err
 }
+
+func URIWithTLSCertsRemoved(remote *url.URL) *url.URL {
```
Comment (dt) [inline pkg/ccl/streamingccl/streamclient/partitioned_stream_client.go:381]: A pattern which I believe @adityamaru has used in a few places is to make this a single method that "consumes" some of the parameters and returns pair of the config struct built from them and the URI now without them. Not sure if that'd fit here but I sorta like its conceptual tidiness versus having to mentally pair two separate methods.

File: pkg/ccl/streamingccl/streamclient/partitioned_stream_client.go (hunk 1 above)
Comment (stevendanna) [reply pkg/ccl/streamingccl/streamclient/partitioned_stream_client.go]: Nope, I originally was going to have this live in some utils package, but then decided to just put it here. Thanks for noting it.

File: pkg/ccl/streamingccl/streamclient/partitioned_stream_client.go (hunk 2 above)
Comment (stevendanna) [reply pkg/ccl/streamingccl/streamclient/partitioned_stream_client.go]: Cool, I was going back and forth on this.  I pushed _something_ like this. It isn't as clean as I might like, but I wanted all of the certificate processing errors to happen up front.

**Review by dt — APPROVED** [review]

**Comment by stevendanna** [comment]
bors r=dt

**Pull request merged**